module_utils/kafka_lib_acl.py

```diff
@@ -18,64 +18,145 @@
-              acl_permission
-            ),
-            pattern_type=ACLPatternType.from_name(acl_pattern_type),
-            name=name,
-            principal=acl_principal,
-            host=acl_host
-        )
-
         acl_resource_found = manager.describe_acls(
             acl_resource, api_version
         )
 
-        if state == 'present':
-            if not acl_resource_found:
-                if not module.check_mode:
-                    manager.create_acls([acl_resource], api_version)
-                changed = True
-                msg += 'successfully created.'
-        elif state == 'absent':
-            if acl_resource_found:
-                if not module.check_mode:
-                    manager.delete_acls([acl_resource], api_version)
-                changed = True
-                msg += 'successfully deleted.'
-
+        acls_marked_present = [ACLResource(
+            resource_type=ACLResourceType.from_name(acl['acl_resource_type']),
+            operation=ACLOperation.from_name(acl['acl_operation']),
+            permission_type=ACLPermissionType.from_name(
+                acl['acl_permission']
+            ),
+            pattern_type=ACLPatternType.from_name(acl['acl_pattern_type']),
+            name=acl['name'],
+            principal=acl['acl_principal'],
+            host=acl['acl_host']
+        ) for acl in acls if acl['state'] == 'present']
+        acls_marked_absent = [ACLResource(
+            resource_type=ACLResourceType.from_name(acl['acl_resource_type']),
+            operation=ACLOperation.from_name(acl['acl_operation']),
+            permission_type=ACLPermissionType.from_name(
+                acl['acl_permission']
+            ),
+            pattern_type=ACLPatternType.from_name(acl['acl_pattern_type']),
+            name=acl['name'],
+            principal=acl['acl_principal'],
+            host=acl['acl_host']
+        ) for acl in acls if acl['state'] == 'absent']
+
+        acls_to_add = []
+        for acl in acls_marked_present:
+            found = False
+            for existing_acl in acl_resource_found:
+                if acl == existing_acl:
+                    found = True
+                    break
+            if not found:
+                acls_to_add.append(acl)
+        acls_to_delete = []
+        for acl in acls_marked_absent:
+            found = False
+            for existing_acl in acl_resource_found:
+                if acl == existing_acl:
+                    found = True
+                    break
+            if found:
+                acls_to_delete.append(acl)
+        # Cleanup others acls
+        if mark_others_as_absent:
+            for existing_acl in acl_resource_found:
+                found = False
+                for acl in acls_marked_present:
+                    if acl == existing_acl:
+                        found = True
+                        break
+                if not found:
+                    acls_to_delete.append(existing_acl)
```

Comment:
```suggestion
        if mark_others_as_absent:
            acls_to_delete.extend(
                [acl for acl in acl_resource_found if acl not in acls_marked_present + acls_marked_absent]
            )
```
I changed for `acls_marked_present + acls_marked_absent` to avoid trying to delete 2 times the same ACL

Reply:
Done